frontend/js/tests/user/Dashboard.test.tsx

```diff
@@ -50,773 +48,755 @@ const props: ListensProps = {
   already_reported_user: false,
 };
 
-const APIService = new APIServiceClass("foo");
-const currentUser = { id: 1, name: "iliekcomputers", auth_token: "fnord" };
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      retry: false,
+    },
+  },
+});
 
-const propsOneListen = {
-  ...recentListensPropsOneListen,
-};
+const queryKey = ["dashboard", {}, {}];
 
-const getComponent = (componentProps: ListensProps) => (
-  <BrowserRouter>
-    <Listens />
-  </BrowserRouter>
+const reactQueryWrapper = ({ children }: any) => (
+  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
 );
 
-// eslint-disable-next-line jest/no-disabled-tests
-xdescribe("Listens page", () => {
-  jest.setTimeout(10000);
+const currentUser = { id: 1, name: "iliekcomputers", auth_token: "fnord" };
+
+let mockSearchParam = {};
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  useSearchParams: () => {
+    const [params, setParams] = React.useState(
+      new URLSearchParams(mockSearchParam)
+    );
+    return [
+      params,
+      (newParams: typeof mockSearchParam) => {
+        mockSearchParam = newParams;
+        setParams(new URLSearchParams(newParams));
+      },
+    ];
+  },
+}));
 
-  let userEventSession: UserEvent;
+describe("Listens page", () => {
+  jest.setTimeout(10000);
+  let server: SetupServerApi;
   beforeAll(async () => {
-    userEventSession = await userEvent.setup();
-    fetchMock.enableMocks();
-    fetchMock.mockIf(
-      (input) => input.url.endsWith("/listen-count"),
-      () => {
-        return Promise.resolve(JSON.stringify({ payload: { count: 42 } }));
+    const handlers = [
+      http.post("/", async (path) => HttpResponse.json(props)),
+      http.get("/1/user/*/following", async (path) =>
+        HttpResponse.json({ following: [] })
+      ),
+      http.get("/1/user/*/followers", async (path) =>
+        HttpResponse.json({ followers: [] })
+      ),
+      http.get("/1/user/iliekcomputers/similar-users", async (path) => {
+        HttpResponse.json({ payload: [] });
+      }),
+      http.get("/1/user/*/listen-count", async (path) =>
+        HttpResponse.json({ payload: { count: 42 } })
+      ),
+      http.post("/1/delete-listen", async (path) => HttpResponse.json({})),
+      http.get("/1/user/*/similar-to/*", async (path) =>
+        HttpResponse.json({ payload: [] })
+      ),
+      http.get("/1/stats/user/*/artists", async (path) =>
+        HttpResponse.json({ payload: [] })
+      ),
+    ];
+    server = setupServer(...handlers);
+    server.listen();
+  });
+  afterEach(() => {
+    queryClient.cancelQueries();
+    queryClient.clear();
```

Comment:
No idea if his is the reason for the weird issue with the test passing on some machines and not others, but `cancelQueries` here returns a Promise and we should probably await it:
```suggestion
  afterEach(async () => {
    await queryClient.cancelQueries();
    queryClient.clear();
```